packages/smart-wallet/src/smartWallet.js

```diff
@@ -60,161 +64,323 @@ const { details: X, quote: q } = assert;
-      purseBalances.set(purse, balance);
-    }
-    updatePublishKit.publisher.publish({
-      updated: 'balance',
-      currentAmount: balance,
-    });
+  const preciousState = {
+    // Invitation makers yielded by offer results
+    offerToInvitationMakers: makeScalarBigMapStore('invitation makers', {
+      durable: true,
+    }),
+    // What purses have reported on construction and by getCurrentAmountNotifier updates.
+    purseBalances: makeScalarMapStore(),
+    // Private purses. This assumes one purse per brand, which will be valid in MN-1 but not always.
+    brandPurses: makeScalarBigMapStore('brand purses', { durable: true }),
   };
 
-  // #endregion
+  const nonpreciousState = {
+    brandDescriptors: makeScalarMapStore(),
+    // To ensure every offer ID is unique we require that each is a number greater
+    // than has ever been used. This high water mark is sufficient to track that.
+    lastOfferId: 0,
+    updatePublishKit: harden(
+      makeStoredPublishKit(myWalletStorageNode, shared.publicMarshaller),
+    ),
+  };
 
-  // #region issuer management
-  /**
-   * Private purses. This assumes one purse per brand, which will be valid in MN-1 but not always.
-   *
-   * @type {MapStore<Brand, RemotePurse>}
-   */
-  const brandPurses = makeScalarBigMapStore('brand purses', { durable: true });
+  return {
+    ...shared,
+    ...unique,
+    ...nonpreciousState,
+    ...preciousState,
+  };
+};
 
-  /** @type { (desc: Omit<BrandDescriptor, 'displayInfo'>, purse: RemotePurse) => Promise<void>} */
-  const addBrand = async (desc, purseRef) => {
-    // assert haven't received this issuer before.
-    const descriptorsHas = brandDescriptors.has(desc.brand);
-    const pursesHas = brandPurses.has(desc.brand);
-    assert(
-      !(descriptorsHas && pursesHas),
-      'repeated brand from bank asset subscription',
-    );
-    assert(
-      !(descriptorsHas || pursesHas),
-      'corrupted state; one store has brand already',
-    );
+const behaviorGuards = {
+  // xxx updateBalance string not really optional. not exposed so okay to skip guards.
+  // helper: M.interface('helperFacetI', {
+  //   addBrand: M.call(
+  //     {
+  //       brand: BrandShape,
+  //       issuer: IssuerShape,
+  //       petname: M.string(),
+  //     },
+  //     PurseShape,
+  //   ).returns(M.promise()),
+  //   updateBalance: M.call(PurseShape, AmountShape, M.opt(M.string())).returns(),
+  // }),
+  deposit: M.interface('depositFacetI', {
+    receive: M.callWhen(M.await(M.eref(PaymentShape))).returns(AmountShape),
+  }),
+  offers: M.interface('offers facet', {
+    executeOffer: M.call(shape.OfferSpec).returns(M.promise()),
+    getLastOfferId: M.call().returns(M.number()),
+  }),
+  self: M.interface('selfFacetI', {
+    handleBridgeAction: M.call(shape.StringCapData, M.boolean()).returns(
+      M.promise(),
+    ),
+    getDepositFacet: M.call().returns(M.eref(M.any())),
+    getOffersFacet: M.call().returns(M.eref(M.any())),
```

Comment:
These facets should be simple to describe.

Reply:
what's the value? these are input guards. why runtime validate the return?

seems like busy work, especially when we have to write types twice (until https://github.com/Agoric/agoric-sdk/issues/6160)